docs/design/enviroment_management.md

```diff
@@ -0,0 +1,156 @@
+# Kusion Environment Management Proposal
+
+## Context
+
+Towards the version 0.10.0, Kusion is about to make a huge upgrade: the unambiguous definition of responsibility and collaboration paradigm between platform engineers and application developers through the application delivery process. The platform engineers firstly define modules and corresponding application-oriented schemas, where the latter constitute the application schema [AppConfiguration](https://www.kusionstack.io/docs/user_docs/config-walkthrough/overview#appconfiguration-model), and secondly initialize environments and corresponding configurations. The application developers select environment and fill in the fields of AppConfiguration. Then, executing the delivery action in a trusted environment, e.g. a CD pipeline, to apply the application intent from both the platform engineers and application developers. 
+
+The upgrade explicitly the role of platform engineers and the concepts of **Module** and **Environment**. The Module is a building block enabling the delivery of a set of resources from **Schema Configuration** and **Environment Configuration**. While the Environment is a collection of application-independent configurations, and an environment configuration is composed of a set of module inputs, which forms the entire module inputs combined with schema configuration.
+
+<br />![module](module.jpg)
+
+## Goals and Non-Goals
+
+### Goals
+
+1. Define the writing format of environment configurations;
+2. Provide the commands to manage environments.
+
+### Non-Goals
+
+1. Only support local file based environment management, the support of other storage backends is not the goal of version 0.10.0. 
+
+## Proposal
+
+### Writing Format
+
+The configuration of an environment should be maintained in a single yaml file, and the writing format and an example is shown as below.
+
+```yaml
+# Module input, with format standard：
+# # module_<ModuleName>:
+# #   default: # default configurations, applied to all projects
+# #     <ModuleInput_Field1>: <value1>
+# #     <ModuleInput_Field2>: <value2>
+# #     ...
+# #   <overrideGroupName>: # override configurations, applied to the projects assigned in projectSelector
+# #     <ModuleInput_Field1>: <value1_Override>
+# #     ... 
+module_Database:
```

Comment:
Why should the module block name have a `module` prefix?  It is also awkward that the module block's name is in the snake format and the `projectSelector` is in the camel format.